tools/install_external.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+# Copyright European Organization for Nuclear Research (CERN)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0OA
+#
+# Authors:
+# - Wen Guan, <wen.guan@cern.ch>, 2016
+
+
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+ROOT = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+PIP_REQUIRES = os.path.join(ROOT, 'tools', 'pip-requires')
+PIP_REQUIRES_TEST = os.path.join(ROOT, 'tools', 'pip-requires-test')
+
+
+def die(message, *args):
+    print >> sys.stderr, message % args
+    sys.exit(1)
+
+
+def run_command(cmd, redirect_output=True, check_exit_code=True, shell=False):
+    """
+    Runs a command in an out-of-process shell, returning the
+    output of that command.  Working directory is ROOT.
+    """
+    if redirect_output:
+        stdout = subprocess.PIPE
+    else:
+        stdout = None
+
+    proc = subprocess.Popen(cmd, cwd=ROOT, stdout=stdout, shell=shell)
+    output = proc.communicate()[0]
+    if check_exit_code and proc.returncode != 0:
+        die('Command "%s" failed.\n%s', ' '.join(cmd), output)
+    return output
+
+
+HAS_PIP = bool(run_command(['which', 'pip'], check_exit_code=False).strip())
+HAS_WGET = bool(run_command(['which', 'wget'], check_exit_code=False).strip())
+
+
+def configure_git():
+    """
+    Configure git to add git hooks
+    """
+    print "Configure git"
+    run_command("%s/tools/configure_git.sh" % ROOT, shell=True)
+
+
+def install_pip():
+    """
+    Install pip to tools/externals
+    """
+    print "Installing pip via wget"
+    if not HAS_WGET:
+        die("ERROR: wget not found, please install.")
+
+    tempdir = tempfile.mkdtemp()
+    run_command(['wget', '-O', os.path.join(tempdir, 'get-pip.py'), 'https://bootstrap.pypa.io/get-pip.py'])
+    run_command("python %s/get-pip.py --root=%s --ignore-installed" % (tempdir, os.path.join(ROOT, 'tools/externals')), shell=True)
```

Comment:
See this line?
It is longer then this view. With 120 symbols, it will fit perfectly, with one spare symbol.